demos/classification_benchmark_demo/cpp/main.cpp

```diff
@@ -118,229 +121,225 @@ cv::Mat centerSquareCrop(const cv::Mat& image) {
+                    if (predictedClass == classificationImageMetaData.groundTruthId) {
+                        predictionResult = PredictionResult::Correct;
+                        correctPredictionsCount++;
+                        label = classificationResult.topLabels[i].label;
+                        break;
+                    }
                 }
-                const ClassificationImageMetaData& classificationImageMetaData
-                    = classificationResult.metaData->asRef<const ClassificationImageMetaData>();
-
-                auto outputImg = classificationImageMetaData.img;
-
-                if (outputImg.empty()) {
-                    throw std::invalid_argument("Renderer: image provided in metadata is empty");
+            } else {
+                predictionResult = PredictionResult::Unknown;
+            }
+            framesNum++;
+            gridMat.updateMat(outputImg, label, predictionResult);
+            accuracy = static_cast<double>(correctPredictionsCount) / framesNum;
+            gridMat.textUpdate(metrics, classificationResult.metaData->asRef<ImageMetaData>().timeStamp, accuracy, FLAGS_nt, isTestMode,
+                                !FLAGS_gt.empty(), presenter);
+            renderMetrics.update(renderingStart);
+            elapsedSeconds = std::chrono::steady_clock::now() - startTime;
+            if (FLAGS_show) {
+                cv::imshow("classification_demo", gridMat.outImg);
+                //--- Processing keyboard events
+                int key = cv::pollKey();
+                if (27 == key || 'q' == key || 'Q' == key) {  // Esc
+                    keepRunning = false;
+                }
+                else if (32 == key || 'r' == key || 'R' == key) {  // press space or r to restart testing if needed
+                    isTestMode = true;
+                    framesNum = 0;
+                    framesNumOnCalculationStart = 0;
+                    correctPredictionsCount = 0;
+                    accuracy = 0;
+                    elapsedSeconds = std::chrono::steady_clock::duration(0);
+                    startTime = std::chrono::steady_clock::now();
                 }
-                PredictionResult predictionResult = PredictionResult::Incorrect;
-                std::string label = classificationResult.topLabels.front().label;
-                if (!FLAGS_gt.empty()) {
-                    for (size_t i = 0; i < FLAGS_nt; i++) {
-                        unsigned predictedClass = classificationResult.topLabels[i].id;
-                        if (predictedClass == classificationImageMetaData.groundTruthId) {
-                            predictionResult = PredictionResult::Correct;
-                            correctPredictionsCount++;
-                            label = classificationResult.topLabels[i].label;
+                else if ('p' == key || 'P' == key) {
+                    while (true) {
```

Comment:
Add `0` and ` ` keys.

You should put pause handling first similar to https://github.com/openvinotoolkit/open_model_zoo/blob/272d2d7001cefc712337f9cde2e2ca02b6b3c923/demos/interactive_face_detection_demo/cpp/main.cpp#L302
That will allow to to remove `while (true)`.

Reply:
Spacebar key is already used for restarting - should I reassign it or add just "0"?

Reply:
Yes, this will align demos

Reply:
Done